pyanaconda/modules/common/structures/network.py

```diff
@@ -116,13 +116,13 @@ def set_from_nm_device(self, device):
         :param device: NetworkManager device object
         :type device: NMDevice
         """
-        self._device_name = device.get_iface()
-        self._device_type = device.get_device_type()
+        self._device_name = device.get_iface() or ""
+        self._device_type = device.get_device_type() or self.DEVICE_TYPE_UNKNOWN
         try:
-            hw_address = device.get_permanent_hw_address()
+            perm_hw_address = device.get_permanent_hw_address()
         except AttributeError:
-            hw_address = device.get_hw_address()
-        self.hw_address = hw_address
+            perm_hw_address = None
```

Comment:
It could be interesting to have a log here.

Reply:
Virtual devices don't have permanent hw address so I think this would produce a lot of useless info.

Reply:
OK. Thanks for info.